### ISSUE TITLE
Creating a webdataset tar file

### ISSUE THREAD

**Issue by KavyaPuranik** [webdataset/webdataset#195]
Hi,

I'm using the webdataset tar file and trying to augment each of the image of the input tarball using the below transformation and write it back to the tarball. You can see the code below.

```
import sys
from torchvision import transforms
import webdataset as wds

normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])


augment = transforms.Compose(
    [
        transforms.RandomResizedCrop(224),
        transforms.RandomHorizontalFlip(),
        transforms.ToTensor(),
        normalize,
    ]
)

def identity(x):
    return x

dataset = wds.WebDataset("-").decode("pil").to_tuple("jpg cls").map_tuple(augment, identity)

sink = wds.TarWriter("-")
for sample in dataset:
    print(sample.get("__key__"), file=sys.stderr)
    processed_sample = {
        "__key__": sample.get("__key__"),
        "jpg": sample.pop("jpg"),
        "cls": sample.pop("cls")
    }
    sink.write(processed_sample)
```

Attaching the sample of input tar ball fyi.

```
root@admin:/# tar -tvf imagenet-train-000000.tar | head  -n 5
-r--r--r-- bigdata/bigdata   3 2020-06-25 17:41 0911032.cls
-r--r--r-- bigdata/bigdata 92227 2020-06-25 17:41 0911032.jpg
-r--r--r-- bigdata/bigdata     3 2020-06-25 17:41 1203092.cls
-r--r--r-- bigdata/bigdata 15163 2020-06-25 17:41 1203092.jpg
-r--r--r-- bigdata/bigdata     3 2020-06-25 17:41 0403282.cls
```

Whenever I try to read the transformed tarball  it throws the below error(This is obvious because the size of the generated tarball is 0)
```
root@admin:/# tar -tvf result3.tar | head -n 5
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
```

Any insights on how to resolve this issue?

**Comment by KavyaPuranik** [comment]
`import sys
from torchvision import transforms
import webdataset as wds
normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
augment = transforms.Compose(
    [
        transforms.RandomResizedCrop(224),
        transforms.RandomHorizontalFlip(),
        transforms.ToTensor(),
        normalize,
    ]
)
def identity(x):
    return x
dataset = wds.WebDataset("-").decode("pil").to_tuple("jpg cls").map_tuple(augment, identity)
sink = wds.TarWriter("-")
for sample in dataset:
    print(sample.get("__key__"), file=sys.stderr)
    processed_sample = {
        "__key__": sample.get("__key__"),
        "jpg": sample.pop("jpg"),
        "cls": sample.pop("cls")
    }
    sink.write(processed_sample)`

**Comment by KavyaPuranik** [comment]
```import sys
from torchvision import transforms
import webdataset as wds
normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
augment = transforms.Compose(
    [
        transforms.RandomResizedCrop(224),
        transforms.RandomHorizontalFlip(),
        transforms.ToTensor(),
        normalize,
    ]
)
def identity(x):
    return x
dataset = wds.WebDataset("-").decode("pil").to_tuple("jpg cls").map_tuple(augment, identity)
sink = wds.TarWriter("-")
for sample in dataset:
    print(sample.get("__key__"), file=sys.stderr)
    processed_sample = {
        "__key__": sample.get("__key__"),
        "jpg": sample.pop("jpg"),
        "cls": sample.pop("cls")
    }
    sink.write(processed_sample)```

**Comment by KavyaPuranik** [comment]
There was an issue in my code,  which I fixed & was able to generate the tar. Hence closing it.

**Comment by danielajisafe** [comment]
is there an easy way to read this ```.jpg``` and ```.cls``` files easily with webdataset without calling dataloader?
